ui/app/AppLayouts/Profile/controls/ShowcaseDelegate.qml

```diff
@@ -0,0 +1,97 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+import StatusQ.Popups 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+import utils 1.0
+
+StatusDraggableListItem {
+    id: root
+
+    property var showcaseObj
+    property int showcaseVisibility: Constants.ShowcaseVisibility.NoOne
+
+    signal showcaseVisibilityRequested(int value)
+
+    component ShowcaseVisibilityAction: StatusMenuItem {
+        property int showcaseVisibility: Constants.ShowcaseVisibility.NoOne
+        icon.name: ProfileUtils.visibilityIcon(showcaseVisibility)
+        icon.color: Theme.palette.primaryColor1
+    }
+
+    icon.width: 40
+    icon.height: 40
+
+    draggable: true
+    dragAxis: Drag.XAndYAxis
+
+    actions: [
+        StatusRoundButton {
+            icon.name: ProfileUtils.visibilityIcon(root.showcaseVisibility)
+            Layout.preferredWidth: 58
+            Layout.preferredHeight: 28
+            border.width: 1
+            border.color: Theme.palette.directColor7
+            radius: 14
+            highlighted: menuLoader.item && menuLoader.item.opened
+            onClicked: {
+                if (menuLoader.active) {
+                    menuLoader.active = false
+                    return
+                }
+                menuLoader.active = true
+                menuLoader.item.popup(width - menuLoader.item.width, height)
+            }
+
+            ButtonGroup {
+                id: showcaseVisibilityGroup
+                exclusive: true
+                onClicked: function(button) {
+                    root.showcaseVisibilityRequested((button as ShowcaseVisibilityAction).showcaseVisibility)
```

Comment:
hmm, what's this `button as ShowcaseVisibilityAction`, is it supported in QML? 
Isn't it TypeScript syntax?
Haven't seen it before 🤔

Reply:
Yup, a new thing in Qt 5.15; basically a type cast :)